File: src/main/java/wtf/lars/breakpointnotify/listener/DebuggerListener.java
package wtf.lars.breakpointnotify.listener;

import com.intellij.ide.DataManager;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.project.Project;
import com.intellij.xdebugger.XDebugProcess;
import com.intellij.xdebugger.XDebuggerManagerListener;
import org.jetbrains.annotations.NotNull;
import wtf.lars.breakpointnotify.service.NotificationService;

public class DebuggerListener implements XDebuggerManagerListener {

    private final NotificationService NOTIFICATION_SERVICE = new NotificationService();

    @Override
    public void processStarted(@NotNull XDebugProcess debugProcess) {
        final Project originProject = debugProcess.getSession().getProject();

        DataManager.getInstance().getDataContextFromFocusAsync().onSuccess(dataContext -> {
            final Project activeProject = dataContext.getData(PlatformDataKeys.PROJECT);

            if (originProject != activeProject) {
                NOTIFICATION_SERVICE.sendNotification(activeProject, originProject);
            }
        });
    }

}

File: src/main/kotlin/wtf/lars/breakpointnotify/listeners/DebuggerListener.kt
package wtf.lars.breakpointnotify.listeners

import com.intellij.ide.DataManager
import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.actionSystem.PlatformDataKeys
import com.intellij.xdebugger.XDebugProcess
import com.intellij.xdebugger.XDebuggerManagerListener
import wtf.lars.breakpointnotify.services.NotificationService

class DebuggerListener : XDebuggerManagerListener {
    private val NOTIFICATION_SERVICE = NotificationService()
    override fun processStarted(debugProcess: XDebugProcess) {
        val originProject = debugProcess.session.project
        DataManager.getInstance().dataContextFromFocusAsync.onSuccess { dataContext: DataContext ->
            val activeProject = dataContext.getData(PlatformDataKeys.PROJECT)
            if (originProject !== activeProject) {
                NOTIFICATION_SERVICE.sendNotification(activeProject, originProject)
            }
        }
    }
}